ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/userActivityServiceAgent.service.ts

```diff
@@ -21,11 +21,18 @@ export class UserActivityServiceAgent extends UserServiceAgent<Array<ActivityIns
     }
 
     public getActivities(studyGuid: Observable<string | null>, participantGuid?: string): Observable<Array<ActivityInstance> | null> {
-        if (participantGuid) {
-            this.session.setParticipant(participantGuid);
-        }
-        return studyGuid.pipe(
-            mergeMap(x => x ? this.getObservable(`/studies/${x}/activities`)
-                : of(null)));
+        return studyGuid.pipe(mergeMap(x => {
+            if (x) {
+                if (participantGuid) {
+                    this.session.setParticipant(participantGuid);
+                }
+                const activitiesObservable = this.getObservable(`/studies/${x}/activities`);
+                if (participantGuid) {
+                    this.session.setParticipant(null);
+                }
+                return activitiesObservable;
```

Comment:
I don't think this is good place to change any non-local state, particularly on the session, which is kind of global. Please find an alternative fix to problem.

Reply:
public getActivities(studyGuid: Observable<string | null>, participantGuid?: string): Observable<Array<ActivityInstance> | null> {
        return studyGuid.pipe(mergeMap(x => {
            if (x) {
                const previousParticipantGuid = this.session.participantGuid;
                if (participantGuid) {
                    this.session.setParticipantGuid(participantGuid);
                }
                try {
                const activitiesObservable = this.getObservable(`/studies/${x}/activities`);
               } finally {
                 this.session.setPartcipantGuid(previousParticipantGuid);
                }
                return activitiesObservable;
}
            return of(null);
        }));

Reply:
done! thanks for the help!